alerts/alerts_test.go

```diff
@@ -0,0 +1,129 @@
+package alerts
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/webhooks"
+	"go.uber.org/zap"
+)
+
+func TestWebhooks(t *testing.T) {
+	mgr := webhooks.NewManager(zap.NewNop().Sugar())
+	alerts := NewManager(mgr)
+
+	mux := http.NewServeMux()
+	var events []webhooks.Action
+	var mu sync.Mutex
+	mux.HandleFunc("/events", func(w http.ResponseWriter, r *http.Request) {
+		var event webhooks.Action
+		if err := json.NewDecoder(r.Body).Decode(&event); err != nil {
+			t.Fatal(err)
+		}
+		mu.Lock()
+		events = append(events, event)
+		mu.Unlock()
+	})
+	srv := httptest.NewServer(mux)
+	defer srv.Close()
+
+	// register a hook
+	wh := webhooks.Webhook{
+		Module: webhookModule,
+		URL:    fmt.Sprintf("http://%v/events", srv.Listener.Addr().String()),
+	}
+	if hookID := wh.String(); hookID != fmt.Sprintf("%v.%v.%v", wh.URL, wh.Module, "") {
+		t.Fatalf("wrong result for wh.String(): %v != %v", wh.String(), hookID)
+	}
+	err := mgr.Register(wh)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// perform some actions that should trigger the endpoint
+	a := Alert{
+		ID:        types.Hash256{1},
+		Message:   "test",
+		Severity:  SeverityWarning,
+		Timestamp: time.Unix(0, 0),
+		Data: map[string]interface{}{
+			"origin": "foo",
```

Comment:
I don't really mind but is there a reason for not making `Origin` a field on the `Alert` type?

Reply:
To keep the response type consistent with `hostd`'s.